pkgs/by-name/na/nautilus-typeahead/patch.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  stdenvNoCC,
+  git,
+  cacert,
+  nautilus,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+
+  pname = "nautilus-typeahead";
+  version = "46.1-1"; # labeled by the AUR package version
+  rev = "f5f593bf36c41756a29d5112a10cf7ec70b8eafb";
+  outputHash = "sha256-z+JX8z/NCrGeizghYfLMpoXCoSjnUzRKjM7vVA1J/zM=";
+
+  name = "${finalAttrs.pname}-${finalAttrs.version}.patch";
+
+  nativeBuildInputs = [ git ];
+  env.GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+  buildCommand = ''
+    set -x
+
+    git clone --filter=blob:none --single-branch \
+      https://gitlab.gnome.org/albertvaka/nautilus.git
+
+    cd nautilus
+    git fetch origin "${finalAttrs.rev}"
+    git remote add upstream \
+      https://gitlab.gnome.org/GNOME/nautilus.git
+
+    git fetch --filter=blob:none --tags upstream
+    git format-patch "${nautilus.version}".."${finalAttrs.rev}" --output="$out"
```

Comment:
This is a very dirty way of doing this. And I am not sure if we wanna even want this fork in Nixpkgs. CC @NixOS/gnome

Reply:
I do not want to maintain this and each Nautilus bump will break it.

Reply:
> This is a very dirty way of doing this.

I understand that this is far from ideal. The reason is, unlike github, AFAIK gitlab cannot generate a `.patch` or `.diff` file which we can `fetchpatch`. That is the reason for doing this manually in `patch.nix`.

> I do not want to maintain this and each Nautilus bump will break it.

Yes I understand. I have stated in the PR,

> As an unofficial package, `nautilus-typeahead` is obviously prone to breakages due to upstream API changes. I have hence set myself as the sole maintainer of this package. In particular, gnome and nautilus maintainers need not be responsible for possible (likely) future breakages of this package. However, due to its popularity, I do believe it would be nice to have it in nixpkgs.

Moreover, I am fine for this PR to be closed if it proves to be a maintenance burden. Simply leaving this in the nixpkgs closed PR list may help others who want the functionality like me.

Reply:
Main issue I see is that the FOD is coupled to `nautilus.version` attribute. So you might not even get a build failure, you might get a FOD mismatch or a possibly incorrect patch. This kind of spooky action at a distance is confusing and cursed.

You could mitigate it by adding `assert nautilus.version == removeRev finalAttrs.version;` to `outputHash` or including `nautilus.version` in the name.

Reply:
> Main issue I see is that the FOD is coupled to `nautilus.version` attribute. So you might not even get a build failure, you might get a FOD mismatch or a possibly incorrect patch. This kind of spooky action at a distance is confusing and cursed.
> 
> You could mitigate it by adding `assert nautilus.version == removeRev finalAttrs.version;` to `outputHash` or including `nautilus.version` in the name.

Indeed, this could be rather dangerous. I have hence decoupled the patch and the nixpkgs nautilus. Instead of `assert`ing I have added a `lib.warnIf` when there is a version mismatch. See the latest commit.